src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.flink.bigquery;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.catalog.CatalogTable;
+import org.apache.flink.table.catalog.CatalogTableImpl;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.table.factories.DynamicTableSourceFactory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.junit.Test;
+
+public class BigQueryDynamicTableFactoryTest {
+  @Test
+  public void factoryIdentifierTestSuccess() {
+    BigQueryDynamicTableFactory bigQueryDynamicTableFactory = new BigQueryDynamicTableFactory();
+    String factoryIdentfier = bigQueryDynamicTableFactory.factoryIdentifier();
+    String expectedFactoryIdentifier = "bigquery";
+    assertThat(factoryIdentfier).isNotNull();
+    assertThat(factoryIdentfier).isEqualTo(expectedFactoryIdentifier);
+    assertTrue(factoryIdentfier instanceof String);
+    assertTrue(bigQueryDynamicTableFactory instanceof DynamicTableSourceFactory);
+  }
+
+  @Test
+  public void requiredOptionsTestSuccess() {
+
+    BigQueryDynamicTableFactory bigQueryDynamicTableFactory = new BigQueryDynamicTableFactory();
+    List<String> expectedOptions =
+        Arrays.asList(
+            "materializationDataset", "materializationProject", "query", "selectedFields", "table");
+    Set<ConfigOption<?>> configOptionSet = bigQueryDynamicTableFactory.requiredOptions();
+    assertThat(configOptionSet).isNotNull();
+    assertTrue(configOptionSet instanceof Set);
```

Comment:
redundant as it is checked by the compiler

Reply:
Removed